components/region_cache_memory_engine/src/write_batch.rs

```diff
@@ -57,9 +61,17 @@ impl RangeCacheWriteBatch {
         Ok(())
     }
 
+    // todo(SpadeA): now, we cache all keys even for those that will not be written
+    // in to the memory engine.
     fn write_impl(&mut self, seq: u64) -> Result<()> {
+        self.engine.handle_pending_load();
+        let mut keys_to_cache: BTreeMap<CacheRange, Vec<(u64, RangeCacheWriteBatchEntry)>> =
+            BTreeMap::new();
         let (engine, filtered_keys) = {
             let core = self.engine.core().read().unwrap();
+            self.buffer
+                .iter()
+                .for_each(|e| e.maybe_cached(seq, &core, &mut keys_to_cache));
```

Comment:
maybe it can be further optimized by fast checking if there's any need to call maybe_cached by just checking ranges_loading_snapshot and ranges_loading_cached_write is empty or not. 
The check can be a help function of range_manager. 
It's faster than iterate the whole buffer in normal cases.

Reply:
Good suggestion.